website/content/docs/connect/dataplane/telemetry.mdx

```diff
@@ -0,0 +1,92 @@
+---
+layout: docs
+page_title: Consul Dataplane - Enable Telemetry Metrics
+description: >-
+  Configure consul-dataplane telemetry to collect metrics you can use to debug and observe Consul behavior and performance.
+---
+
+# Consul Dataplane Telemetry
+
+Consul Dataplane collects metrics about its own status and performance.
+The following external metrics stores are supported:
+
+- [DogstatsD](https://docs.datadoghq.com/developers/dogstatsd/)
+- [Prometheus](https://prometheus.io/docs/prometheus/latest/)
+- [StatsD](https://github.com/statsd/statsd)
+
+# Configuring Telemetry
+
+<Tabs>
+
+<Tab heading="Kubernetes">
+
+Refer to the [Consul on Kubernetes documentation](/docs/k8s/connect/observability/metrics) to
+configure telemetry for Consul on Kubernetes.
+
+</Tab>
+
+<Tab heading="Reference Documentation">
```

Comment:
This is kind of goofy, so open to changing this.

Kubernetes is the only platform where consul-dataplane is supported for now. So I "hid" all of the configuration details in this second tab that I called "Reference Configuration", so that it's obvious if you are actually using consul-dataplane on k8s that you shouldn't care about these details and instead should go straight to the k8s docs to configure telemetry.

I don't think this great as-is, but I do like the tab structure so that, when we do get to supporting consul-dataplane on other platforms, we can add tabs to link to or describe telemetry config for those platforms.

Reply:
@pglass - Sorry, but I'm confused. We only support Consul Dataplane on Kubernetes, but we're documenting information that users cannot put into action? I think we should stash the unusable documentation for now until non-Kubernetes environments are supported.

Reply:
>  I think we should stash the unusable documentation for now until non-Kubernetes environments are supported.

I'm fine with this 🙂 

Elaborating a bit: I guess my view is that consul-dataplane is a component that Consul Kubernetes uses, and that it is useful to have component-level documentation on how to use consul-dataplane. Because as much as we strive to have a great UX, occasionally (or maybe often) someone needs to dig under the hood to debug or change things, and reference docs are pretty useful for that.

So, I see the reference docs for enabling telemetry here being potentially useful, but certainly rarely used and not the first content people should reach for when using consul-dataplane (for now).

Reply:
Now hold on a minute, so, a user _could_ follow the instructions under the reference documentation tab? Is it that the primary use case is to configure telemetry settings through Kubernetes and the secondary use case is to configure telemetry at through the Consul client? If its is something like that, then we _should_ keep the information and describe it it terms of primary and secondary use-cases.

Reply:
How about we just document all the telemetry-* CLI flags under the CLI reference page so that anyone needing more consul-dataplane specific fine tuning options still has a place to goto - looks like everything under `Additional Prometheus Options` has already been specified in the CLI reference, so we can remove the section here?

We can keep this page more aligned with kubernetes configuration with some details about how things work under the hood? So basically, most of the things under the reference section need not be a separate tab, but just a section to highlight further details about the wiring.

Reply:
> Now hold on a minute, so, a user could follow the instructions under the reference documentation tab? Is it that the primary use case is to configure telemetry settings through Kubernetes and the secondary use case is to configure telemetry at through the Consul client?

I wouldn't say it's a secondary use case. If I'm using Kubernetes I _shouldn't_ have to care about the specific flags passed to consul-dataplane, unless I'm encountering unexpected issues and need to debug or etc.

> So basically, most of the things under the reference section need not be a separate tab, but just a section to highlight further details about the wiring.

👍 I simplified the telemetry page, and instead elaborated on a few of the prometheus settings in the CLI reference page. Let me know what you think. I opted not to put any platform-specific callouts here.